### PR TITLE
#165273422 Enable users to receive real time notifications

### PR DESCRIPTION
**Pull request by KabohaJeanMark**
**What does this PR do?**

- Create push notifications app
- use Django Pusher to send these notifications in real time via WebSockets
- Use Django signals to trigger and save notifications for authors when users do the following:
1. Like or dislike their article
2. Favourite their article
3. Comment on their article
4. Follow them

**How to test it out?**
As an authenticated user, either like, dislike, favourite or comment on an article.
While running your server, check out the page on localhost root directory and view the instant notification. 

A glimpse at real-time notification received when an article is liked. It informs the author of the particular article title

Screen Shot 2019-05-14 at 6 35 24 PM

[#165273422](https://www.pivotaltracker.com/story/show/165273422)

### CODE REVIEW

File: authors/apps/pushnotifications/models.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from django.conf import settings
+
+from pusher import Pusher
+
+from authors.apps.articles.models import Article, Like_Dislike
+from authors.apps.favorites. models import Favorites
+from authors.apps.comments.models import Comment
+from authors.apps.authentication.models import User
+
+pusher = Pusher(app_id=u'779166', key=u'931e65a8378bb81125c3',
```
Comment (mwinel) [inline authors/apps/pushnotifications/models.py]: Is it possible for these to be added to the `.env`?

**Review by mwinel — CHANGES_REQUESTED** [review]

File: authors/apps/pushnotifications/models.py (hunk 1 above)
Comment (KabohaJeanMark) [reply authors/apps/pushnotifications/models.py]: Yes, @mwinel. You were very right. I incorporated your feedback in my new changes as I refactored and added some more functionality in the recent changes

**Review by mwinel — APPROVED** [review]

**Comment by mwinel** [comment]
@archibishop This LGTM.

**Review by engjames — APPROVED** [review]
LGTM

**Review by 3Nakajugo — APPROVED** [review]

**Review by archibishop — APPROVED** [review]
LGTM @KabohaJeanMark I am merging this.

**Pull request merged**